src/components/table-toolbar/KeycloakDataTable.tsx

```diff
@@ -242,26 +242,58 @@ export function KeycloakDataTable<T>({
   );
 
   const _onSelect = (isSelected: boolean, rowIndex: number) => {
+
     if (rowIndex === -1) {
-      setRows(
-        rows!.map((row) => {
-          row.selected = isSelected;
-          return row;
-        })
-      );
+
+      if (filteredData) {
+        setRows(
+          filteredData!.map((row) => {
+            row.selected = isSelected;
+            return row;
+          })
+        );
+      } else {
+        setRows(
+          rows!.map((row) => {
+            row.selected = isSelected;
+            return row;
+          })
+        );
+        setRows(rows);
+      }
```

Comment:
I think this can be simplified to:

```suggestion
   setRows(
        (filteredData || rows).map((row) => {
          row.selected = isSelected;
          return row;
        })
      );
```